wa/commands/runstatus.py

```diff
@@ -0,0 +1,298 @@
+import os
+import logging
+from datetime import datetime, timedelta
+
+import colorama
+
+from wa.framework.configuration.core import Status
+from wa import Command, settings
+from wa.framework.output import RunOutput, discover_wa_outputs
+from wa.framework.exception import NotFoundError
+from wa.utils.terminalsize import get_terminal_size
+from wa.utils.doc import format_column
+from wa.utils.log import COLOR_MAP, RESET_COLOR
+
+class StatusCommand(Command):
+
+    name = 'runstatus'
+    description = '''
+    Monitor ongoing runs and provide information on their progress.
+
+    '''
+
+    def initialize(self, context):
+        self.parser.add_argument('-d', '--directory',
+                                 help='''
+                                 Specify the WA output path. runstatus will
+                                 otherwise attempt to discover output
+                                 directories in the current directory.
+                                 ''')
+
+    def execute(self, config, args):
+        if args.directory:
+            output_path = args.directory
+            run_output = RunOutput(output_path)
+        else:
+            possible_outputs = list(discover_wa_outputs(os.getcwd()))
+            num_paths = len(possible_outputs)
+
+            if num_paths > 1:
+                print('More than one possible output directory found,'
+                      ' please choose a path from the following:'
+                      )
+
+                for i in range(num_paths):
+                    print(f"{i}: {possible_outputs[i].basepath}")
+
+                while True:
+                    try:
+                        select = int(input())
+                    except ValueError:
+                        print(f"Please select number from "
+                              f"{[x for x in range(num_paths)]}"
+                              )
+                        continue
+
+                    if select not in range(num_paths):
+                        print(f"Please select number from "
+                              f"{[x for x in range(num_paths)]}"
+                              )
+                        continue
+                    else:
+                        break
+
+                run_output = possible_outputs[select]
+
+            else:
+                run_output = possible_outputs[0]
+
+        rs = RunMonitor(run_output)
```

Comment:
Why `rs`? Wouldn't `rm` (or, even better, `monitor`) make more sense?

Reply:
Ah, old name for the class was RunState before I realise how bad that was.